docs/root/configuration/runtime.rst

```diff
@@ -79,6 +79,29 @@ old tree to the new runtime tree, using the equivalent of the following command:
 
 It's beyond the scope of this document how the file system data is deployed, garbage collected, etc.
 
+Using runtime overrides for deprecated features
+-----------------------------------------------
+
+The Envoy runtime is also a part of the Envoy feature deprecation process.
+
+As described in the Envoy `breaking change policy <//github.com/envoyproxy/envoy/blob/master/CONTRIBUTING.md#breaking-change-policy>`_,
+feature deprecation in Envoy is in 3 phases: warn-by-default, fail-by-default, and code removal.
+
+In the first phase, Envoy simply logs a warning to the warning log that the feature is deprecated.
+Users are encouraged to go to
+`DEPRECATED.md <https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md>` to see how to
+migrate to the new code path and make sure it is suitable for their use case.
+
+In the second phase the message and filename will be added to
+`runtime_features.h <https://github.com/envoyproxy/envoy/blob/master/source/common/runtime/runtime_features.h>`
+and use of that configuration field will cause the config to be rejected by default. 
+This fail-by-default mode can be overridden in runtime configuration. For a proto styled
+``Foo.Bar.Eep`` in ``baz.proto`` by setting the runtime value
+``baz.proto:Eep true``. Use of this override is **strongly discouraged**.
```

Comment:
@alyssawilk not sure if this is ready for rereview or not, thoughts on:

> Do we want to use a common prefix here to avoid collisions?

Reply:
Oh sorry, I had a prefix prefix.field and killed it off when I made it file.proto:field  I think that's unambiguous enough but I can go back to prefixing if you think it's worthwhile.

Reply:
I think I would prefer a prefix if that is OK with you, because then all of the overrides will be in a common directory/tree no matter what the file name is?

Reply:
Sure thing!  As we're not using runtime yet I was unsure of convention.
Reinstated envoy.deprecated_features but feel free to suggest something different